src/lib.rs

```diff
@@ -182,26 +182,34 @@ impl iter::FromIterator<char> for SmolStr {
     }
 }
 
+fn build_from_str_iter<T>(mut iter: impl Iterator<Item=T>) -> SmolStr
+where
+    T: AsRef<str>,
+    std::string::String: std::iter::Extend<T>,
```

Comment:
Why not just `String: iter::Extend<T>`?

Reply:
Because I like pasting in what rustc tell me 🤷‍♂ (fixed)